### DISCUSSION TITLE
移除解析数据用的已弃用函数

### DISCUSSION THREAD

**Discussion by mark9804** [Arondight/Adachi-BOT#550] [Ideas]
前端用来解析 url 的 `escape()` 也被标记成 deprecated 了

https://github.com/Arondight/Adachi-BOT/blob/1970a26f01e591c752c8e8247407335d2011bc03/src/views/component/card-8/app.js#L97-L99

我试了一下直接换成 [encodeURIComponent()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent)，看上去是正常的，就是不知道有没有什么隐藏的雷，不知道老大哥了不了解这个

**Comment by Arondight** [comment]
不了解，我只粘贴复制的，明天再改改

**Reply by Arondight** [reply]
我看了一眼，发现不会改，你提代码的时候顺手帮忙改改吧……

**Reply by Arondight** [reply]
开发文档里面也对应改改

**Reply by mark9804** [reply]
那我等其他迁移做完了一起提，现在也不是不能用

**Comment by mark9804** [comment]
插个开发文档口袋锚点备忘

https://github.com/Arondight/Adachi-BOT/blob/1671a4604cacbea7b36bceb3c370187477527df3/docs/%E5%BC%80%E5%8F%91%E6%8C%87%E5%BC%95.md?plain=1#L185-L195

https://github.com/Arondight/Adachi-BOT/blob/master/docs/%E5%BC%80%E5%8F%91%E6%8C%87%E5%BC%95.md

**Comment by mark9804** [comment]
痛苦起来了，看起来要想办法自己实现一个 atob

https://stackoverflow.com/a/30106551

**Reply by mark9804** [reply]
我知道了，data 里面如果有中文就会歇逼

<img src="https://user-images.githubusercontent.com/9006264/148045738-3d468cb0-16eb-4956-bee3-5e8fc0ac3d49.png" alt="image" height="640">

**Reply by mark9804** [reply]
```javascript
function decodeUnicode(str) {
  // Going backwards: from bytestream, to percent-encoding, to original string.
  return decodeURIComponent(atob(str).split('').map(function (c) {
    return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2);
  }).join(''));
}
```

https://attacomsian.com/blog/javascript-base64-encode-decode
